Got Internal Server Error when publish extension

Please help, I met the following error when publish a new version of my extension

` ovsx publish --pat <token> .\qt-support-Kylin-IDE-0.0.5.vsix`
❌  Internal Server Error
See the documentation for more information:
https://github.com/eclipse/openvsx/wiki/Publishing-Extensions

Publisher: KylinIdeTeam
name: qt-support
version 0.0.5

Note: got errors below trying to publish. Also tried web page, but dialogs just spins forever after clicking 'Publish'.
First time in several months that I've done a Publish. Worked last time I tried.

Note: *** means redacted

$ npx ovsx publish ***.vsix --debug --pat ***
Error: Internal Server Error
    at IncomingMessage.<anonymous> (/usr/lib/node_modules/ovsx/lib/registry.js:185:40)
    at IncomingMessage.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
npm notice 
npm notice New major version of npm available! 9.5.1 -> 10.8.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.8.1
npm notice Run npm install -g npm@10.8.1 to update!
npm notice

-Note:using sudo b/c in the past node-related installed/updates seem to run into permissions errors for me if I do not.

$  sudo npm install -g npm@10.8.1
[sudo] password for ***:             
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: npm@10.8.1
npm ERR! notsup Not compatible with your version of node/npm: npm@10.8.1
npm ERR! notsup Required: {"node":"^18.17.0 || >=20.5.0"}
npm ERR! notsup Actual:   {"npm":"9.5.1","node":"v18.16.1"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2024-06-21T18_25_54_919Z-debug-0.log

Note: I've uninstalled npm  and re-installed from linux package manager; version will be too old, per messages, but I can try  to update again.
Note: Also had to reinstall ovsx module

$ npx ovsx publish ***.vsix --debug --pat ***
Need to install the following packages:
  ovsx
Ok to proceed? (y) y
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'ovsx@0.9.1',
npm WARN EBADENGINE   required: { node: '>= 16' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@vscode/vsce@2.29.0',
npm WARN EBADENGINE   required: { node: '>= 16' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'tmp@0.2.3',
npm WARN EBADENGINE   required: { node: '>=14.14' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/identity@4.3.0',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'cockatiel@3.1.3',
npm WARN EBADENGINE   required: { node: '>=16' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/core-auth@1.7.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/core-client@1.9.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/core-rest-pipeline@1.16.0',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/core-tracing@1.1.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/core-util@1.9.0',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/logger@1.1.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/msal-node@2.9.2',
npm WARN EBADENGINE   required: { node: '>=16' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/abort-controller@2.1.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/abort-controller@2.1.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/abort-controller@2.1.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'http-proxy-agent@7.0.2',
npm WARN EBADENGINE   required: { node: '>= 14' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'https-proxy-agent@7.0.4',
npm WARN EBADENGINE   required: { node: '>= 14' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@azure/abort-controller@2.1.2',
npm WARN EBADENGINE   required: { node: '>=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'agent-base@7.1.1',
npm WARN EBADENGINE   required: { node: '>= 14' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm WARN deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported

Note: retrying upgrade

$ npm install -g npm@10.8.1
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: npm@10.8.1
npm ERR! notsup Not compatible with your version of node/npm: npm@10.8.1
npm ERR! notsup Required: {"node":"^18.17.0 || >=20.5.0"}
npm ERR! notsup Actual:   {"npm":"8.5.1","node":"v12.22.9"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/***/.npm/_logs/2024-06-21T19_06_15_299Z-debug-0.log

Note: back to where I started, but with even older npm/node


Sys Info:
System:
  Kernel: 5.15.0-112-generic x86_64 bits: 64 compiler: gcc v: 11.4.0 Desktop: Cinnamon 6.0.4
    tk: GTK 3.24.33 wm: muffin vt: 7 dm: LightDM 1.30.0 Distro: Linux Mint 21.3 Virginia
    base: Ubuntu 22.04 jammy
Machine:
  Type: Desktop System: Dell product: OptiPlex 5040 v: N/A serial: <superuser required> Chassis:
    type: 3 serial: <superuser required>
  Mobo: Dell model: 0R790T v: A00 serial: 
<superuser required> UEFI-[Legacy]: Dell v: 1.2.7
    date: 01/15/2016
Battery:
  Device-1: hidpp_battery_0 model: Logitech M570 serial: <filter> charge: 40% rechargeable: yes
    status: Discharging
CPU:
  Info: quad core model: Intel Core i5-6500 bits: 64 type: MCP smt: <unsupported> arch: Skylake-S
    rev: 3 cache: L1: 256 KiB L2: 1024 KiB L3: 6 MiB
  Speed (MHz): avg: 3196 high: 3226 min/max: 800/3600 cores: 1: 3153 2: 3210 3: 3196 4: 3226
    bogomips: 25599
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: NVIDIA GP108 [GeForce GT 1030] driver: nvidia v: 535.171.04 pcie: speed: 8 GT/s
    lanes: 4 ports: active: none off: HDMI-A-1 empty: DVI-D-1 bus-ID: 01:00.0 chip-ID: 10de:1d01
    class-ID: 0300
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: nvidia
    unloaded: fbdev,modesetting,nouveau,vesa gpu: nvidia display-ID: :0 screens: 1
  Screen-1: 0 s-res: 3840x2160 s-dpi: 162 s-size: 602x341mm (23.7x13.4") s-diag: 692mm (27.2")
  Monitor-1: HDMI-0 res: 3840x2160 hz: 60 dpi: 163 size: 600x340mm (23.6x13.4")
    diag: 690mm (27.2")
  OpenGL: renderer: NVIDIA GeForce GT 1030/PCIe/SSE2 v: 4.6.0 NVIDIA 535.171.04
    direct render: Yes
Audio:
  Device-1: Intel 100 Series/C230 Series Family HD Audio vendor: Dell driver: snd_hda_intel
    v: kernel bus-ID: 00:1f.3 chip-ID: 8086:a170 class-ID: 0403
  Device-2: NVIDIA GP108 High Definition Audio driver: snd_hda_intel v: kernel pcie:
    speed: 8 GT/s lanes: 4 bus-ID: 01:00.1 chip-ID: 10de:0fb8 class-ID: 0403
  Sound Server-1: ALSA v: k5.15.0-112-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: Intel Ethernet I219-V vendor: Dell driver: e1000e v: kernel port: N/A bus-ID: 00:1f.6
    chip-ID: 8086:15b8 class-ID: 0200
  IF: enp0s31f6 state: up speed: 1000 Mbps duplex: full mac: <filter>
Drives:
  Local Storage: total: 2.05 TiB used: 1.8 TiB (87.7%)
  ID-1: /dev/sda vendor: SanDisk model: SD8SBAT-256G-1006 size: 238.47 GiB speed: 6.0 Gb/s
    type: SSD serial: <filter> rev: 1006 scheme: GPT
  ID-2: /dev/sdb vendor: Western Digital model: WDS100T2B0A-00SM50 size: 931.51 GiB
    speed: 6.0 Gb/s type: SSD serial: <filter> rev: 20WD scheme: MBR
  ID-3: /dev/sdd type: USB vendor: Western Digital model: WD My Passport 25E1 size: 931.48 GiB
    type: N/A serial: <filter> rev: 1021 scheme: GPT
Partition:
  ID-1: / size: 233.62 GiB used: 78.04 GiB (33.4%) fs: ext4 dev: /dev/sda3
  ID-2: /boot/efi size: 47.2 MiB used: 6 MiB (12.8%) fs: vfat dev: /dev/sda2
  ID-3: /home size: 915.82 GiB used: 614.21 GiB (67.1%) fs: ext4 dev: /dev/sdb1
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 768 KiB (0.0%) priority: -2 file: /swapfile
USB:
  Hub-1: 1-0:1 info: Hi-speed hub with single TT ports: 16 rev: 2.0 speed: 480 Mb/s
    chip-ID: 1d6b:0002 class-ID: 0900
  Device-1: 1-5:2 info: 816820090226 type: Mass Storage driver: usb-storage interfaces: 1
    rev: 2.0 speed: 480 Mb/s power: 500mA chip-ID: 3207:0300 class-ID: 0806 serial: <filter>
  Device-2: 1-8:3 info: Logitech Unifying Receiver type: Keyboard,Mouse,HID
    driver: logitech-djreceiver,usbhid interfaces: 3 rev: 2.0 speed: 12 Mb/s power: 98mA
    chip-ID: 046d:c52b class-ID: 0300
  Hub-2: 2-0:1 info: Super-speed hub ports: 10 rev: 3.0 speed: 5 Gb/s chip-ID: 1d6b:0003
    class-ID: 0900
  Device-1: 2-5:2 info: Western Digital My Passport (WD20NMVW) type: Mass Storage
    driver: usb-storage interfaces: 1 rev: 3.1 speed: 5 Gb/s power: 896mA chip-ID: 1058:25e1
    class-ID: 0806 serial: <filter>
Sensors:
  System Temperatures: cpu: 56.0 C pch: 66.5 C mobo: 27.8 C gpu: nvidia temp: 50 C
  Fan Speeds (RPM): N/A gpu: nvidia fan: 87%
Repos:
  Packages: note: see --pkg apt: 4186
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/gambas-team-gambas3-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/gambas-team-gambas3-jammy.gpg] https: //ppa.launchpadcontent.net/gambas-team/gambas3/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/jonaski-strawberry-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/jonaski-strawberry-jammy.gpg] https: //ppa.launchpadcontent.net/jonaski/strawberry/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/krupkat-code-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/krupkat-code-jammy.gpg] https: //ppa.launchpadcontent.net/krupkat/code/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com virginia main upstream import backport
    2: deb http: //archive.ubuntu.com/ubuntu jammy main restricted universe multiverse
    3: deb http: //archive.ubuntu.com/ubuntu jammy-updates main restricted universe multiverse
    4: deb http: //archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ jammy-security main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/vivaldi.list
    1: deb [arch=amd64] https: //repo.vivaldi.com/stable/deb/ stable main
  Active apt repos in: /etc/apt/sources.list.d/vscode.list
    1: deb [arch=amd64,arm64,armhf] http: //packages.microsoft.com/repos/code stable main
  Active apt repos in: /etc/apt/sources.list.d/vscodium.list
    1: deb [ signed-by=/usr/share/keyrings/vscodium-archive-keyring.gpg ] https: //download.vscodium.com/debs vscodium main
  Active apt repos in: /etc/apt/sources.list.d/warpdotdev.list
    1: deb [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/warpdotdev.gpg] https: //releases.warp.dev/linux/deb stable main
Info:
  Processes: 322 Uptime: 9h 17m wakeups: 21 Memory: 15.54 GiB used: 6.39 GiB (41.1%) Init: systemd
  v: 249 runlevel: 5 Compilers: gcc: 11.4.0 alt: 11/12 clang: 14.0.0-1ubuntu1.1
  Client: Unknown python client inxi: 3.3.13

[2024-06-21T18_24_39_065Z-debug-0.log](https://github.com/user-attachments/files/15932349/2024-06-21T18_24_39_065Z-debug-0.log)
[2024-06-21T18_30_42_235Z-debug-0.log](https://github.com/user-attachments/files/15932350/2024-06-21T18_30_42_235Z-debug-0.log)
[2024-06-21T18_40_58_608Z-debug-0.log](https://github.com/user-attachments/files/15932351/2024-06-21T18_40_58_608Z-debug-0.log)
[2024-06-21T18_41_08_144Z-debug-0.log](https://github.com/user-attachments/files/15932352/2024-06-21T18_41_08_144Z-debug-0.log)
[2024-06-21T18_41_54_108Z-debug-0.log](https://github.com/user-attachments/files/15932354/2024-06-21T18_41_54_108Z-debug-0.log)
[2024-06-21T18_45_49_193Z-debug-0.log](https://github.com/user-attachments/files/15932355/2024-06-21T18_45_49_193Z-debug-0.log)
[2024-06-21T19_03_42_499Z-debug-0.log](https://github.com/user-attachments/files/15932356/2024-06-21T19_03_42_499Z-debug-0.log)
[2024-06-21T19_05_55_108Z-debug-0.log](https://github.com/user-attachments/files/15932357/2024-06-21T19_05_55_108Z-debug-0.log)
[2024-06-21T19_06_15_299Z-debug-0.log](https://github.com/user-attachments/files/15932358/2024-06-21T19_06_15_299Z-debug-0.log)

--- a/server/src/main/java/org/eclipse/openvsx/migration/GenerateKeyPairJobRequestHandler.java
+++ b/server/src/main/java/org/eclipse/openvsx/migration/GenerateKeyPairJobRequestHandler.java
@@ -63,19 +63,16 @@ public class GenerateKeyPairJobRequestHandler implements JobRequestHandler<Handl
 
     private void createKeyPair() {
         var activeKeyPair = repositories.findActiveKeyPair();
-        Streamable<ExtensionVersion> extVersions;
         if(activeKeyPair == null) {
-            service.generateKeyPair();
-            extVersions = repositories.findVersions();
+            renewKeyPair();
         } else {
-            extVersions = repositories.findVersionsWithout(activeKeyPair);
+            repositories.findVersionsWithout(activeKeyPair).forEach(this::enqueueCreateSignatureJob);
         }
-
-        extVersions.forEach(this::enqueueCreateSignatureJob);
     }
 
     private void renewKeyPair() {
-        service.renewKeyPair();
+        var keyPair = service.generateKeyPair();
+        service.updateKeyPair(keyPair);
         repositories.findVersions().forEach(this::enqueueCreateSignatureJob);
     }
 

--- a/server/src/main/java/org/eclipse/openvsx/migration/GenerateKeyPairJobService.java
+++ b/server/src/main/java/org/eclipse/openvsx/migration/GenerateKeyPairJobService.java
@@ -20,6 +20,7 @@ import org.bouncycastle.util.io.pem.PemObject;
 import org.bouncycastle.util.io.pem.PemWriter;
 import org.eclipse.openvsx.entities.SignatureKeyPair;
 import org.eclipse.openvsx.repositories.RepositoryService;
+import org.eclipse.openvsx.util.TimeUtil;
 import org.springframework.stereotype.Component;
 
 import java.io.ByteArrayOutputStream;
@@ -27,7 +28,6 @@ import java.io.IOException;
 import java.io.OutputStreamWriter;
 import java.nio.charset.StandardCharsets;
 import java.security.SecureRandom;
-import java.time.LocalDateTime;
 import java.util.UUID;
 
 @Component
@@ -42,17 +42,12 @@ public class GenerateKeyPairJobService {
     }
 
     @Transactional
-    public void renewKeyPair() {
-        var activeKeyPair = repositories.findActiveKeyPair();
-        if(activeKeyPair != null) {
-            activeKeyPair.setActive(false);
-        }
-
-        generateKeyPair();
+    public void updateKeyPair(SignatureKeyPair keyPair) {
+        repositories.deactivateKeyPairs();
+        entityManager.persist(keyPair);
     }
 
-    @Transactional
-    public void generateKeyPair() {
+    public SignatureKeyPair generateKeyPair() {
         var generator = new Ed25519KeyPairGenerator();
         generator.init(new Ed25519KeyGenerationParameters(new SecureRandom()));
         var pair = generator.generateKeyPair();
@@ -61,9 +56,9 @@ public class GenerateKeyPairJobService {
         keyPair.setPublicId(UUID.randomUUID().toString());
         keyPair.setPrivateKey(((Ed25519PrivateKeyParameters) pair.getPrivate()).getEncoded());
         keyPair.setPublicKeyText(getPublicKeyText(pair));
-        keyPair.setCreated(LocalDateTime.now());
+        keyPair.setCreated(TimeUtil.getCurrentUTC());
         keyPair.setActive(true);
-        entityManager.persist(keyPair);
+        return keyPair;
     }
     
     private String getPublicKeyText(AsymmetricCipherKeyPair pair) {

--- a/server/src/main/java/org/eclipse/openvsx/migration/MigrationScheduler.java
+++ b/server/src/main/java/org/eclipse/openvsx/migration/MigrationScheduler.java
@@ -15,7 +15,9 @@ import org.springframework.boot.context.event.ApplicationStartedEvent;
 import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Component;
 
+import java.nio.charset.StandardCharsets;
 import java.time.Instant;
+import java.util.UUID;
 
 @Component
 public class MigrationScheduler {
@@ -25,6 +27,9 @@ public class MigrationScheduler {
     @Value("${ovsx.migrations.delay.seconds:0}")
     long delay;
 
+    @Value("${ovsx.registry.version:}")
+    String registryVersion;
+
     public MigrationScheduler(JobRequestScheduler scheduler) {
         this.scheduler = scheduler;
     }
@@ -32,6 +37,8 @@ public class MigrationScheduler {
     @EventListener
     public void applicationStarted(ApplicationStartedEvent event) {
         var instant = Instant.now().plusSeconds(delay);
-        scheduler.schedule(instant, new HandlerJobRequest<>(MigrationRunner.class));
+        var jobIdText = "MigrationRunner::" + registryVersion;
+        var jobId = UUID.nameUUIDFromBytes(jobIdText.getBytes(StandardCharsets.UTF_8));
+        scheduler.schedule(jobId, instant, new HandlerJobRequest<>(MigrationRunner.class));
     }
 }

--- a/server/src/main/java/org/eclipse/openvsx/publish/ExtensionVersionIntegrityService.java
+++ b/server/src/main/java/org/eclipse/openvsx/publish/ExtensionVersionIntegrityService.java
@@ -156,9 +156,12 @@ public class ExtensionVersionIntegrityService {
             zip.stream()
                     .filter(entry -> !entry.isDirectory())
                     .forEach(entry -> {
-                        var content = ArchiveUtil.readEntry(zip, entry, ObservationRegistry.NOOP);
-                        var manifestEntry = generateManifestEntry(content, mapper, base64);
-                        manifestEntries.set(new String(base64.encode(entry.getName().getBytes(StandardCharsets.UTF_8))), manifestEntry);
+                        try {
+                            var manifestEntry = generateManifestEntry(zip.getInputStream(entry).readAllBytes(), mapper, base64);
+                            manifestEntries.set(new String(base64.encode(entry.getName().getBytes(StandardCharsets.UTF_8))), manifestEntry);
+                        } catch (IOException e) {
+                            logger.warn("Failed to add entry to manifest", e);
+                        }
                     });
         }
 

--- a/server/src/main/java/org/eclipse/openvsx/repositories/RepositoryService.java
+++ b/server/src/main/java/org/eclipse/openvsx/repositories/RepositoryService.java
@@ -609,4 +609,8 @@ public class RepositoryService {
     public List<String> findExtensionTargetPlatforms(Extension extension) {
         return extensionVersionJooqRepo.findDistinctTargetPlatforms(extension);
     }
+
+    public void deactivateKeyPairs() {
+        signatureKeyPairRepo.updateActiveSetFalse();
+    }
 }

--- a/server/src/main/java/org/eclipse/openvsx/repositories/SignatureKeyPairRepository.java
+++ b/server/src/main/java/org/eclipse/openvsx/repositories/SignatureKeyPairRepository.java
@@ -10,6 +10,8 @@
 package org.eclipse.openvsx.repositories;
 
 import org.eclipse.openvsx.entities.SignatureKeyPair;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.Repository;
 
 public interface SignatureKeyPairRepository extends Repository<SignatureKeyPair, Long> {
@@ -19,4 +21,8 @@ public interface SignatureKeyPairRepository extends Repository<SignatureKeyPair,
     void deleteAll();
 
     SignatureKeyPair findByPublicId(String publicId);
+
+    @Modifying
+    @Query("update SignatureKeyPair k set k.active = false")
+    void updateActiveSetFalse();
 }

--- a/server/src/main/java/org/eclipse/openvsx/storage/StorageMigration.java
+++ b/server/src/main/java/org/eclipse/openvsx/storage/StorageMigration.java
@@ -23,6 +23,8 @@ import org.springframework.stereotype.Component;
 import org.springframework.transaction.support.TransactionTemplate;
 import org.springframework.web.client.RestTemplate;
 
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
@@ -95,7 +97,8 @@ public class StorageMigration {
                             + migrations.get(i) + " to " + storageType + ".");
                 }
             }
-            scheduledFuture = taskScheduler.scheduleWithFixedDelay(this::migrateResources, migrationDelay);
+            var duration = Duration.of(migrationDelay, ChronoUnit.MILLIS);
+            scheduledFuture = taskScheduler.scheduleWithFixedDelay(this::migrateResources, duration);
         }
     }
 

--- a/server/src/test/java/org/eclipse/openvsx/publish/ExtensionVersionIntegrityServiceTest.java
+++ b/server/src/test/java/org/eclipse/openvsx/publish/ExtensionVersionIntegrityServiceTest.java
@@ -12,15 +12,16 @@ package org.eclipse.openvsx.publish;
 import io.micrometer.observation.ObservationRegistry;
 import jakarta.persistence.EntityManager;
 import org.eclipse.openvsx.cache.CacheService;
-import org.eclipse.openvsx.entities.*;
+import org.eclipse.openvsx.entities.Extension;
+import org.eclipse.openvsx.entities.ExtensionVersion;
+import org.eclipse.openvsx.entities.FileResource;
+import org.eclipse.openvsx.entities.Namespace;
 import org.eclipse.openvsx.migration.GenerateKeyPairJobService;
 import org.eclipse.openvsx.repositories.RepositoryService;
 import org.eclipse.openvsx.util.ArchiveUtil;
 import org.eclipse.openvsx.util.TempFile;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Mockito;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.TestConfiguration;
 import org.springframework.boot.test.mock.mockito.MockBean;
@@ -34,11 +35,8 @@ import java.util.zip.ZipFile;
 import static org.junit.jupiter.api.Assertions.*;
 
 @ExtendWith(SpringExtension.class)
-@MockBean({ CacheService.class, RepositoryService.class })
+@MockBean({ CacheService.class, RepositoryService.class, EntityManager.class })
 public class ExtensionVersionIntegrityServiceTest {
-
-    @MockBean
-    EntityManager entityManager;
 
     @Autowired
     ExtensionVersionIntegrityService integrityService;
@@ -48,10 +46,7 @@ public class ExtensionVersionIntegrityServiceTest {
 
     @Test
     public void testGenerateSignature() throws IOException {
-        keyPairService.generateKeyPair();
-        var keyPairCaptor = ArgumentCaptor.forClass(SignatureKeyPair.class);
-        Mockito.verify(entityManager).persist(keyPairCaptor.capture());
-        var keyPair = keyPairCaptor.getValue();
+        var keyPair = keyPairService.generateKeyPair();
 
         var namespace = new Namespace();
         namespace.setName("foo");

--- a/server/src/test/java/org/eclipse/openvsx/repositories/RepositoryServiceSmokeTest.java
+++ b/server/src/test/java/org/eclipse/openvsx/repositories/RepositoryServiceSmokeTest.java
@@ -202,7 +202,8 @@ class RepositoryServiceSmokeTest {
                 () -> repositories.findLatestVersion(extension, "targetPlatform", false, false),
                 () -> repositories.findLatestVersions(namespace),
                 () -> repositories.findLatestVersions(userData),
-                () -> repositories.findExtensionTargetPlatforms(extension)
+                () -> repositories.findExtensionTargetPlatforms(extension),
+                () -> repositories.deactivateKeyPairs()
         );
 
         // check that we did not miss anything

